content/docs/functions/json_build_object.md

```diff
@@ -0,0 +1,265 @@
+| 3  | Sneakers   | 49.99 | Sporty sneakers for everyday use  | Footwear | {"size": "10", "color": "Black", "rating": 4.2}
+```
+
+{/*
+```sql
+CREATE TABLE products_with_rating (
+   id SERIAL PRIMARY KEY,
+   name TEXT NOT NULL,
+   price DECIMAL(5, 2) NOT NULL,
+   description TEXT,
+   category TEXT,
+   attributes JSONB
+);
+
+
+INSERT INTO products_with_rating (name, price, description, category, attributes)
+VALUES
+   ('T-Shirt', 25.99, 'A comfortable cotton T-Shirt', 'Clothing', json_build_object(
+       'color', 'Blue',
+       'size', 'Medium',
+       'rating', 4.5
+   )),
+   ('Coffee Mug', 12.99, 'A ceramic mug with a funny design', 'Kitchen', json_build_object(
+       'color', 'White',
+       'size', 'Small',
+       'rating', 3.8
+   )),
+   ('Sneakers', 49.99, 'Sporty sneakers for everyday use', 'Footwear', json_build_object(
+       'color', 'Black',
+       'size', '10',
+       'rating', 4.2
+   ));
+```
+*/}
+
+
+Retrieve the products, building a JSON structure with `json_build_object` and ordering the results based on the `rating`:
+
+
+```sql
+SELECT
+   id,
+   name,
+   price,
+   json_build_object(
+       'category', category,
+       'description', description,
+       'attributes', json_build_object(
+           'color', attributes->>'color',
+           'size', attributes->>'size',
+           'rating', attributes->>'rating'
+       )
+   ) AS details
+FROM products_with_rating
+ORDER BY (attributes->>'rating')::NUMERIC DESC;
+```
+
+
+`ORDER BY` was to order the results based on the descending order of rating.
+
+
+Returns:
+
+```text
+| id |    name    | price |                                                                        details                                                                       
+|----|------------|-------|-------------------------------------------------------------------------------------------------------------------------------------------------------
+| 1  | T-Shirt    | 25.99 | {"category" : "Clothing", "description" : "A comfortable cotton T-Shirt", "attributes" : {"color" : "Blue", "size" : "Medium", "rating" : "4.5"}}
+| 3  | Sneakers   | 49.99 | {"category" : "Footwear", "description" : "Sporty sneakers for everyday use", "attributes" : {"color" : "Black", "size" : "10", "rating" : "4.2"}}
+| 2  | Coffee Mug | 12.99 | {"category" : "Kitchen", "description" : "A ceramic mug with a funny design", "attributes" : {"color" : "White", "size" : "Small", "rating" : "3.8"}}
+```
+
+### Use with `GROUP BY`
+
+
+To create a JSON object that groups the total price for each category of products in the products table:
+
+
+```sql
+SELECT
+   category,
+   json_build_object(
+       'total_price', sum(price)
+   ) AS category_total_price
+FROM products
+GROUP BY category;
+```
+
+
+Returns:
+
+```text
+| category |  category_total_price  
+|----------|-------------------------
+| Kitchen  | {"total_price" : 12.99}
+| Clothing | {"total_price" : 25.99}
+```
```

Comment:
```suggestion
```
## Resources

- [PostgreSQL Documentation: JSON Functions and Operators](https://www.postgresql.org/docs/current/functions-json.html)

- [PostgreSQL Documentation: JSON Types](https://www.postgresql.org/docs/current/datatype-json.html)
```

Reply:
- We should include a Resources section on all pages as part of the template. 
- We might want to have these links go to our mirrored docs, but would like to see if PixelPoint can do that for us globally somewhere in the build or web server. This would let us turn it off globally later if we move away from mirrored docs.
- Any other relevant links, it would be good to add also.